micrometer/prometheus/src/main/java/io/quarkus/ts/openshift/micrometer/PrimeNumberResource.java

```diff
@@ -0,0 +1,76 @@
+package io.quarkus.ts.openshift.micrometer;
+
+import io.micrometer.core.instrument.MeterRegistry;
+import org.apache.commons.lang3.RandomStringUtils;
+
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+
+import java.util.concurrent.atomic.LongAccumulator;
+import java.util.function.Supplier;
+
+@Path("/")
+public class PrimeNumberResource {
+
+    private final LongAccumulator highestPrime = new LongAccumulator(Long::max, 0);
+    private final MeterRegistry registry;
+    private final String uniqueId = RandomStringUtils.randomAlphabetic(5);
+
+    PrimeNumberResource(MeterRegistry registry) {
+        this.registry = registry;
+
+        // Create a gauge that uses the highestPrimeNumberSoFar method
+        // to obtain the highest observed prime number
+        registry.gauge("prime.number.max." + uniqueId, this,
+                PrimeNumberResource::highestObservedPrimeNumber);
+    }
+
+    @GET
+    @Path("/uniqueId")
+    @Produces(MediaType.TEXT_PLAIN)
+    public String getUniqueId() {
+        return uniqueId;
+    }
+
+    @GET
+    @Path("/check/{number}")
+    @Produces(MediaType.TEXT_PLAIN)
+    public String checkIfPrime(@PathParam("number") long number) {
+        if (number < 1) {
+            return "Only natural numbers can be prime numbers.";
+        }
+        if (number == 1) {
+            return "1 is not prime.";
+        }
+        if (number == 2) {
+            return "2 is prime.";
+        }
+        if (number % 2 == 0) {
+            return number + " is not prime, it is divisible by 2.";
+        }
+
+        Supplier<String> supplier = () -> {
+            for (int i = 3; i < Math.floor(Math.sqrt(number)) + 1; i = i + 2) {
+                if (number % i == 0) {
+                    return number + " is not prime, is divisible by " + i + ".";
+                }
+            }
+            highestPrime.accumulate(number);
+            return number + " is prime.";
+        };
+
+        return registry.timer("prime.number.test." + uniqueId).wrap(supplier).get();
```

Comment:
Add uniqueId as a tag. (sugestion)

Example:

`registry.timer("prime.number" , "DEV", uniqueId)`
`registry.timer("prime.number" , "TEST", uniqueId)`

Reply:
Using a unique id extremely ease the test verification (see how we assert the metrics).